chanbackup/pubsub.go

```diff
@@ -321,6 +370,42 @@ func (s *SubSwapper) backupUpdater() {
 					err)
 			}
 
+		// We received a manual update. Handle it and update the file.
+		case manualUpdate := <-s.manualUpdates:
+			oldStateSize := len(s.backupState)
+
+			// For all open channels, we'll create a new SCB given
+			// the required information.
+			for _, single := range manualUpdate.singles {
+				log.Debugf("Manual update of channel %v",
+					single.FundingOutpoint)
+
+				s.backupState[single.FundingOutpoint] = single
+			}
+
+			newStateSize := len(s.backupState)
+
+			log.Infof("Updating on-disk multi SCB backup: "+
+				"num_old_chans=%v, num_new_chans=%v",
+				oldStateSize, newStateSize)
+
+			// With out new state constructed, we'll, atomically
+			// update the on-disk backup state.
+			err := s.updateBackupFile()
+			if err != nil {
+				log.Errorf("unable to update backup file: %v",
+					err)
+
+				// Send the error to the caller of ManualUpdate.
+				// The error channel must have capacity of 1 not
```

Comment:
Do you mean to make sure that the channel does have capacity of 1? We can send to the channel in select and log an error or `panic` in the default case. What do you think?

Reply:
Could you explain why we cannot block here, backupUpdater runs in its own goroutine, why not sync this call via the errChan with the server func?

Reply:
`ManualUpdate` receives from `errChan` and `s,quit` in `select`. If `s.quit` is closed while waiting, then nobody reads from `errChan` and `backupUpdater` goroutine leaks. This can be worked around by using another `select` to send to `errChan` and receive from `s.quit` at the same time, but I like the approach with a buffered channel more, because it is harder to make a dead-lock this way.

Reply:
thanks for the explantion: 

just do `manualUpdate.errChan <- s.updateBackupFile()` here.

Moreover add a comment why we need to be buffered when creating the errChan in `ManualUpdate`